src/app/shared/services/form-builder.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Store } from '@ngrx/store';
+
+import { FormBuilderModel } from 'src/app/shared/models/form-builder.model';
+import * as actions from 'src/app/core/actions';
+import { NameValueInterface } from 'src/app/shared/models/name-value-interface';
+
+@Injectable({
+  providedIn: 'root'
```

Comment:
please read about root, is it necessary here?

Reply:
it added when created service,  change, add to providers in shared module,